src/httprpc.cpp

```diff
@@ -223,6 +248,13 @@ static bool InitRPCAuthentication()
         LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcuser for rpcauth auth generation.\n");
         strRPCUserColonPass = gArgs.GetArg("-rpcuser", "") + ":" + gArgs.GetArg("-rpcpassword", "");
     }
+
+    for (const std::string& strRPCWhitelist : gArgs.GetArgs("-rpcwhitelist")) {
+        std::string strUser = strRPCWhitelist.substr(0, strRPCWhitelist.find(':'));
+        std::string strWhitelist = strRPCWhitelist.substr(strRPCWhitelist.find(':') + 1);
```

Comment:
This behavior may be fine, but this will silently turn "foo" into `strUser == strWhitelist == "foo"`.

Reply:
Agree, it should check for correct format.